MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,778 @@
+
+Depending on your set of custom modules some migration effort might be necessary.
+Custom modules that are loaded in addition to core GLSP modules should work as before in 2.x without any changes.
+Custom modules that are a fullfledged replacement for a GLSP core feature module need to be migrated to a `FeatureModule`.
+To keep the dependency chain intact the replacement module has to reuse the `featureId` of the base module it is replacing.
+For instance if you have a custom `myViewportModule` that serves as a replacement for the base `viewportModule` than the module
+declaration should look like this:
+
+```ts
+export const myViewportModule = new FeatureModule(
+    (bind, _unbind, isBound) => {
+        /// your custom module configuration
+    },
+    { featureId: viewportModule.featureId }
+);
+```
+
+In addition, the naming of feature modules in 1.0.0 was an inconsistent mix of modules with no prefix, prefixed with `glsp` and prefixed with `sprotty`.
+In 2.x the naming has been aligned and all feature modules provided by GLSP now have no prefix.
+In addition, several modules have been renamed to better reflect the purpose of the module.
+We recommend to use `Search & Replace` to migrate affected module references.
+
+<details>
+  <summary>List of changes</summary>
+
+-   defaultGlspModule-> baseModule
+-   glspExportModule -> exportModule
+-   glspBoundsModule -> boundsModule
+-   glspCommandPaletteModule -> commandPaletteModule
+-   glspContextMenuModule -> contextMenuModule
+-   glspDecorationModule -> decorationModule
+-   glspEditLabelModule -> labelEditModule
+-   glspHoverModule -> hoverModule
+-   glspSelectModule -> selectModule
+-   glspServerCopyPasteModule -> serverCopyPasteModule
+-   glspViewportModule -> viewportModule
+-   modelHintsModule -> typeHintsModule
+-   glspRoutingModule -> routingModule
+-   enableDefaultToolsOnFocusLossModule -> toolFocusLossModule
+
+</details>
+
+### Tools module rework
+
+In 1.0 the default tools where configured with two generic modules (`toolsModule` & `toolFeedbackModule`) that provided the configuration
+for all tools at once.
+This made it quite hard to customize tools individually.
+In addition, the separation into a dedicated `feedbackModule` was not ideal as it split the tool configuration across two interdependent modules.
+With 2.x this has been reworked. The generic `toolsModule` & `toolFeedbackModule` have been removed in favor of individual tool modules:
+
+-   changeBoundsToolModule
+-   deletionToolModule
+-   edgeCreationToolModule
+-   edgeEditToolModule
+-   marqueeSelectionToolModule
+-   nodeCreationToolModule.
+
+So if you are using customized default tools in your project some migration effort is required to adapt to new imports and the new module structure.
+
+<details>
+  <summary>List of changes</summary>
+
+-   GLSPTool -> Tool
+-   <Basetool>.dispatchFeedback -> Basetool>.registerFeedback
```

Comment:
This does not show up properly: BaseTool.dispatchFeedback -> BaseEditTool.registerFeedback or skip the class name entirely.